src/AutoRest.CSharp/LowLevel/Generation/LowLevelRestClientWriter.cs

```diff
@@ -191,19 +192,49 @@ private void WriteStatusCodeSwitch(CodeWriter writer, RestClientMethod clientMet
                             writer.Line($"case int s when s >= {statusCode.Family * 100:L} && s < {statusCode.Family * 100 + 100:L}:");
                         }
                     }
-                    writer.Line($"return {messageVariable}.{nameof(HttpMessage.Response)};");
+                    var returnType = clientMethod.ReturnType;
+                    if (responseBody != null && responseBody is ConstantResponseBody body && returnType != null)
+                    {
+                        writer.Append($"return {typeof(Response)}.FromValue(");
+                        writer.WriteReferenceOrConstant(body.Value);
+                        writer.Append($", {messageVariable}.{nameof(HttpMessage.Response)});");
+                    }
+                    else
+                    {
+                        writer.Line($"return {messageVariable}.{nameof(HttpMessage.Response)};");
+                    }
                 }
 
                 writer.Line($"default:");
-                if (async)
+                if (isNoThrow)
```

Comment:
I'm not familiar with the LLC CodeGen code yet, but it looks strange to me that we changed this if/else statement to use a different condition.  Do we still handle the async/sync case somewhere?

Reply:
Yes, below there are several methods that handle async/sync, but I agree that we need to reformat this a bit.

Reply:
Updated!